src/otx/cli/cli.py

```diff
@@ -0,0 +1,316 @@
+            if default_config_file.exists():
+                with Path(default_config_file).open() as f:
+                    default_config = yaml.safe_load(f)
+                sub_parser.set_defaults(**default_config)
+
+            if "logger" in added:
+                sub_parser.link_arguments("engine.work_dir", "logger.init_args.save_dir")
+            if "callbacks" in added:
+                sub_parser.link_arguments("callback_monitor", "callbacks.init_args.monitor")
+                sub_parser.link_arguments("engine.work_dir", "callbacks.init_args.dirpath")
+
+            self._subcommand_method_arguments[subcommand] = added
+            self._subcommand_parsers[subcommand] = sub_parser
+            parser_subcommands.add_subcommand(subcommand, sub_parser, help=description)
+
+    def _set_extension_subcommands_parser(self, parser_subcommands: _ActionSubCommands) -> None:
+        from otx.cli.install import add_install_parser
+
+        add_install_parser(parser_subcommands)
+
+    def instantiate_classes(self) -> None:
+        """Instantiate the necessary classes based on the subcommand.
+
+        This method checks if the subcommand is one of the engine subcommands.
+        If it is, it instantiates the necessary classes such as config, datamodule, model, and engine.
+        """
+        if self.subcommand in self.engine_subcommands():
+            self.config_init = self.parser.instantiate_classes(self.config)
+            self.datamodule = self._get(self.config_init, "data")
+            self.model, optimizer, scheduler = self.instantiate_model()
+
+            engine_kwargs = self._get(self.config_init, "engine")
+            self.engine = Engine(
+                model=self.model,
+                optimizer=optimizer,
+                scheduler=scheduler,
+                datamodule=self.datamodule,
+                **engine_kwargs,
+            )
+
+    def instantiate_model(self) -> tuple:
+        """Instantiate the model based on the subcommand.
+
+        This method checks if the subcommand is one of the engine subcommands.
+        If it is, it instantiates the model.
+
+        Returns:
+            tuple: The model and optimizer and scheduler.
+        """
+        model = self._get(self.config_init, "model")
+        optimizer_kwargs = self._get(self.config_init, "optimizer", {})
+        if isinstance(optimizer_kwargs, Namespace):
+            optimizer_kwargs = namespace_to_dict(optimizer_kwargs)
+        scheduler_kwargs = self._get(self.config_init, "scheduler", {})
+        if isinstance(scheduler_kwargs, Namespace):
+            scheduler_kwargs = namespace_to_dict(scheduler_kwargs)
+        from otx.core.utils.instantiators import partial_instantiate_class
+
+        return model, partial_instantiate_class(optimizer_kwargs), partial_instantiate_class(scheduler_kwargs)
+
+    def _get(self, config: Namespace, key: str, default: Any = None) -> Any:  # noqa: ANN401
```

Comment:
is this the same method name from jasonargparse that we overwrite

Reply:
how about something like `get_config_value` or `values` or something similar?

Reply:
Thank! renamed it - https://github.com/openvinotoolkit/training_extensions/commit/3679a70296e06f1ed7c5fd6fab1899bf65fcf023